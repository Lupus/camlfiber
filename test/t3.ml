let i = ref 10
let a = Fiber.create (fun () -> while !i > 0 do
				  Printf.printf "a %i\n" !i;
				  decr i;
				  Fiber.yield ();
				done) ()
let b = Fiber.create (fun () -> while !i > 0 do
				  Printf.printf "b %i\n" !i;
				  decr i;
				  Fiber.resume a
				done) ()
let _ =
  Fiber.resume a;
  Fiber.resume b
