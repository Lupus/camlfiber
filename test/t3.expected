a 10
b 9
a 8
b 7
a 6
b 5
a 4
b 3
a 2
b 1
